dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/MessageImpl.java

```diff
@@ -119,72 +126,133 @@
     private static final String RESOURCE_METHOD = "org.apache.cxf.resource.method";
     private static final String ASYNC_RESPONSE = "javax.ws.rs.container.AsyncResponse";
     private static final String SSE_EVENT_SINK = "javax.ws.rs.sse.SseEventSink";
-    private static final Set<String> KEYS;
+    private static final Map<String, Integer> KEYMAP;
+    private static String[] propertyNames = new String[TOTAL];
 
+    private static final Object NOT_FOUND = new Object();
+    private static final Integer KEY_NOT_FOUND = Integer.valueOf(-1);
+    
     static {
-        Set<String> keys = new HashSet<String>();
-        keys.add(CONTENT_TYPE);
-        keys.add(PROTOCOL_HEADERS);
-        keys.add(QUERY_STRING);
-        keys.add(AbstractHTTPDestination.HTTP_REQUEST);
-        keys.add(AbstractHTTPDestination.HTTP_RESPONSE);
-        keys.add(REQUEST_PATH_TO_MATCH_SLASH);
-        keys.add(HTTP_REQUEST_METHOD);
-        keys.add(INTERCEPTOR_PROVIDERS);
-        keys.add(TEMPLATE_PARAMETERS);
-        keys.add(ACCEPT_CONTENT_TYPE);
-        keys.add(CONTINUATION_PROVIDER);
-        keys.add(DESTINATION);
-        keys.add(OP_RES_INFO_STACK);
-        keys.add(WSDL_DESCRIPTION);
-        keys.add(WSDL_INTERFACE);
-        keys.add(WSDL_OPERATION);
-        keys.add(WSDL_PORT);
-        keys.add(WSDL_SERVICE);
-        keys.add(REQUEST_URL);
-        keys.add(REQUEST_URI);
-        keys.add(PATH_INFO);
-        keys.add(BASE_PATH);
-        keys.add(FIXED_PARAMETER_ORDER);
-        keys.add(IN_INTERCEPTORS);
-        keys.add(OUT_INTERCEPTORS);
-        keys.add(RESPONSE_CODE);
-        keys.add(ATTACHMENTS);
-        keys.add(ENCODING);
-        keys.add(AbstractHTTPDestination.HTTP_CONTEXT);
-        keys.add(AbstractHTTPDestination.HTTP_CONFIG);
-        keys.add(AbstractHTTPDestination.HTTP_CONTEXT_MATCH_STRATEGY);
-        keys.add(HTTP_BASE_PATH);
-        keys.add(ASYNC_POST_RESPONSE_DISPATCH);
-        keys.add(SECURITY_CONTEXT);
-        keys.add(AUTHORIZATION_POLICY);
-        keys.add(CERT_CONSTRAINTS);
-        keys.add(AbstractHTTPDestination.SERVICE_REDIRECTION);
-        keys.add(HTTP_SERVLET_RESPONSE);
-        keys.add(RESOURCE_METHOD);
-        keys.add(ONE_WAY_REQUEST);
-        keys.add(ASYNC_RESPONSE);
-        keys.add(THREAD_CONTEXT_SWITCHED);
-        keys.add(OutgoingChainInterceptor.CACHE_INPUT_PROPERTY);
-        keys.add(PhaseInterceptorChain.PREVIOUS_MESSAGE);
-        keys.add(AbstractHTTPDestination.RESPONSE_HEADERS_COPIED);
-        keys.add(SSE_EVENT_SINK);
-        keys.add(REQUESTOR_ROLE);
-        keys.add(PARTIAL_RESPONSE_MESSAGE);
-        keys.add(EMPTY_PARTIAL_RESPONSE_MESSAGE);
-        keys.add(ENDPOINT_ADDRESS);
-        keys.add(INBOUND_MESSAGE);
-        KEYS = Collections.unmodifiableSet(keys);
+        Map<String, Integer> keymap = new HashMap<String, Integer>(TOTAL);
```

Comment:
This can be significantly simplified:

Define your propertyNames during the declaration of the field, i.e.

propertyNames = {CONTENT_TYPE, PROTOCOL_HEADERS, ...

Then in the static initialize you can easily just do a for loop:

for (int i = 0; i < TOTAL; ++i) {
    keymap.put(propertyNames[i], i);
}